src/Common/src/System/Security/Cryptography/KeyFormatHelper.cs

```diff
@@ -497,8 +484,7 @@ private static unsafe void WriteEncodedSpan(AsnWriter writer, ReadOnlySpan<byte>
             }
             finally
             {
-                CryptographicOperations.ZeroMemory(decryptedMemory.Span);
-                ArrayPool<byte>.Shared.Return(decrypted.Array);
+                ArrayPool<byte>.Shared.Return(decrypted.Array, clearArray: true);
```

Comment:
Why?  As decrypted.Length approaches decrypted.Array.Length/2 the simple "clear everything" gets more expensive.

Reply:
I'll revert it. I didn't have the `Span` available anymore, so I had to rewrite it somehow. I didn't realize that `clearArray` could result in much more memory to be cleaned up.

Reply:
`clearArray: true` doesn't know where things were written, so it clears everything.  Span.Clear() or CryptographicOperations.ZeroMemory(Span) both only clear the requested range.

Reply:
It is obvious once you pointed it out. :)